src/json-api/json-api-hapi-plugin-spec.ts

```diff
@@ -1,53 +1,78 @@
 
-import { expect } from 'chai';
+import { expect, use } from 'chai';
 import * as moment from 'moment';
 import * as queryString from 'querystring';
 import 'reflect-metadata';
-
-import * as Hapi from '../server/hapi';
+import * as sinon from 'sinon';
+import * as sinonChai from 'sinon-chai';
+use(sinonChai);
+
+import AuthenticationHapiPlugin from '../authentication/authentication-hapi-plugin';
+import { get, kernel } from '../config';
+import { getTestServer, IServerInjectOptions } from '../server/hapi';
+import { fetchMock } from '../shared/fetch';
 import { MINARD_ERROR_CODE } from '../shared/minard-error';
 import { JsonApiHapiPlugin, parseActivityFilter } from './json-api-hapi-plugin';
 import { JsonApiModule } from './json-api-module';
 
-const provisionServer = async (plugin: JsonApiHapiPlugin) => {
-  const server = Hapi.getServer();
-  server.connection({ port: 8080 });
-  const options = {
-    register: plugin.register,
-    routes: {
+const provisionServer = async (plugin?: JsonApiHapiPlugin) => {
+  const server = await getTestServer(
+    {
+      register: get<AuthenticationHapiPlugin>(AuthenticationHapiPlugin.injectSymbol).registerNoop,
+    },
+    {
+      register: (plugin || get<JsonApiHapiPlugin>(JsonApiHapiPlugin.injectSymbol)).register,
+      routes: {
         prefix: '/api',
+      },
     },
-  };
-  await server.register([options]);
+  );
   return server;
 };
 
 const baseUrl = 'http://localhost:8000';
 
 describe('json-api-hapi-plugin', () => {
 
+  let stubs: sinon.SinonStub[];
+
+  const stubJsonApi = (stubber: (api: JsonApiModule) => sinon.SinonStub | sinon.SinonStub[]) => {
+    const jsonApiModule = get<JsonApiModule>(JsonApiModule.injectSymbol);
```

Comment:
To be specific, I'm not against using DI for testing. However, I would prefer to avoid unnecessarily using a **DI container** for testing.

I think this makes tests easier to reason about, as functions and classes that individual tests depend on are kept explicit.

They can be kept explicit also if using a DI container. But for this to be the case, if using a DI container, I would use a configuration that only sets up the bindings that a particular test requires.